### PR TITLE
Optimize SegmentParseV2::parse code logic

### PR DESCRIPTION
**Pull request by zhaoyuguang**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [ ] New feature provided
- [x] Improve performance
___
### New feature or improvement
- Describe the details and related test reports.
Passing my test `parseBinarySegment(upstreamSegment)` is a time-consuming and stable method that does not reduce time-consuming because of multiple calls.
image

### CODE REVIEW

**Comment by wu-sheng** [comment]
Why the parse method called multiple time? I think this is not true.

**Comment by zhaoyuguang** [comment]
With debug, `bufferData.getV2Segment() == null` is true so parse method called multiple time.

**Comment by wu-sheng** [comment]
Get your point. I am thinking more, if this is the case, why don't we just get v2 segment after if null block?

File: oap-server/server-receiver-plugin/skywalking-trace-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/trace/provider/parser/SegmentParseV2.java [hunk 1]
```diff
@@ -82,10 +82,11 @@ public boolean parse(BufferData<UpstreamSegment> bufferData, SegmentSource sourc
 
             List<UniqueId> traceIds = upstreamSegment.getGlobalTraceIdsList();
 
+            SegmentObject segmentObject = parseBinarySegment(upstreamSegment);
```
Comment (wu-sheng) [inline oap-server/server-receiver-plugin/skywalking-trace-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/trace/provider/parser/SegmentParseV2.java]: `SegmentParseV2#parse` will be invoked multiple times, due to `SegmentParseV2#Producer#call` triggered by `DataStreamReader#L159-L161`. So, the `#parseBinarySegment` should be called only when `bufferData.getV2Segment() == null` from my perspective

Comment (zhaoyuguang) [reply oap-server/server-receiver-plugin/skywalking-trace-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/trace/provider/parser/SegmentParseV2.java]: After the debug, as you said, read from the file and go back to this judgment.

**Pull request merged**